runtime/ast/unmarshal.go

```diff
@@ -0,0 +1,901 @@
+package ast
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/onflow/cadence/runtime/common"
+)
+
+func jsonMarshalAndVerify(i interface{}) (b []byte, err error) {
+	if b, err = json.Marshal(i); err == nil {
+		expected := string(b)
+		roundtripped := reflect.New(reflect.TypeOf(i)).Interface()
+		if err = json.Unmarshal(b, &roundtripped); err == nil {
+			if b, err = json.Marshal(roundtripped); err == nil {
+				actual := string(b)
+				if expected != actual {
+					return nil, fmt.Errorf("un/marshal failed:\n%s\n%s\n", expected, actual)
+				}
+			}
+		}
+	}
```

Comment:
Instead of nesting the code deeper and deeper, consider exiting early to keep the flow flat, e.g
```suggestion
	if b, err = json.Marshal(i); err != nil {
		return nil, err
	}
	expected := string(b)
	roundtripped := reflect.New(reflect.TypeOf(i)).Interface()
	if err = json.Unmarshal(b, &roundtripped); err != nil {
		return nil, err
	}
	if b, err = json.Marshal(roundtripped); err != nil {
		return nil, err
	}
	actual := string(b)
	if expected != actual {
		return nil, fmt.Errorf("un/marshal failed:\n%s\n%s\n", expected, actual)
	}
}
```

Also, given that this is a test-only function, move it to the test files.